tools/notebooks/boosted-metastaking.ipynb

```diff
@@ -0,0 +1,578 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Collect users"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "metadata": {}
+   },
+   "outputs": [],
+   "source": [
+    "from utils.utils_scenarios import collect_farm_contract_users\n",
+    "from multiversx_sdk_network_providers import ApiNetworkProvider\n",
+    "from utils.utils_scenarios import FetchedUser\n",
+    "from typing import List\n",
+    "\n",
+    "mainnet_api = ApiNetworkProvider(\"https://api.multiversx.com\")\n",
+    "fetched_users = collect_farm_contract_users(10, metastake_contract.address, metastake_contract.farm_token, metastake_contract.metastake_token,\n",
+    "                                            mainnet_api, context.network_provider.proxy)\n",
+    "\n",
+    "users: List[FetchedUser] = fetched_users.get_users_with_farm_tokens()\n",
+    "if not users:\n",
+    "    raise Exception('No users found for the given criteria')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "metadata": {}
+   },
+   "outputs": [],
+   "source": [
+    "from utils.decoding_structures import METASTAKE_TOKEN_ATTRIBUTES, FARM_TOKEN_ATTRIBUTES, STAKE_V2_TOKEN_ATTRIBUTES, STAKE_V1_TOKEN_ATTRIBUTES\n",
+    "from utils.utils_chain import decode_merged_attributes, base64_to_hex, get_all_token_nonces_details_for_account, Account\n",
+    "\n",
+    "index = 0\n",
+    "user = Account(pem_file=\"~/Documents/sh1.pem\")\n",
+    "user.address = users[index].address\n",
+    "user.sync_nonce(context.network_provider.proxy)\n",
+    "tokens_in_account = get_all_token_nonces_details_for_account(metastake_contract.farm_token, user.address.bech32(), context.network_provider.proxy)\n",
+    "print(f'Account: {user.address.bech32()}')\n",
+    "print(f'Farm Tokens in account:')\n",
+    "for token in tokens_in_account:\n",
+    "    print(f'\\t{token}')\n",
+    "tokens_in_account = get_all_token_nonces_details_for_account(metastake_contract.metastake_token, user.address.bech32(), context.network_provider.proxy)\n",
+    "\n",
+    "print(f'Metastake Tokens in account:')\n",
+    "for token in tokens_in_account:\n",
+    "    print(f'\\t{token}')\n",
+    "    _, _, _ = metastake_contract.get_all_decoded_metastake_token_attributes_from_proxy(context.network_provider.proxy, user.address.bech32(), token[\"nonce\"])\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "State"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "metadata": {}
+   },
+   "outputs": [],
+   "source": [
+    "from contracts.simple_lock_energy_contract import SimpleLockEnergyContract\n",
+    "from contracts.staking_contract import StakingContract\n",
+    "from contracts.farm_contract import FarmContract\n",
+    "\n",
+    "def get_all_stats(metastake_contract: MetaStakingContract, user: Account):\n",
```

Comment:
it might worth moving this method into metastaking contract implementation

Reply:
yeah... might be useful. Will move this report into contracts (even though get_all_stats is not actually ALL stats... just some).